framework/src/st_things/things_utils/include/things_time.h

```diff
@@ -0,0 +1,64 @@
+#pragma once
+
+#include <things_common.h>
+
+/**
+ * @file things_time.h
+ * @brief time API
+ */
+
+/**
+ * @defgroup time time
+ * @brief <b> time API </b>
+ *
+ * 1. dawit time module
+ *  - dawit use microsecond resolution system time
+ *	- dawit use utc time
+ *
+ * @brief <b> time module example </b>
+ * @includelineno  ex_time.c
+ *
+ * @{
+ */
+
+/**
+* @brief time_info structure
+*/
+typedef struct {
+	int second;		/**<  Seconds  										*/
+	int minute;		/**<  Minutes 										*/
+	int hour;		/**<  Hour (0--23) 									*/
+	int day;		/**<  Day of month (1--31) 							*/
+	int month;		/**<  Month (0--11) 								*/
+	int year;		/**<  Year (calendar year minus 1900) 				*/
+	int day_of_week;/**<  Weekday (0--6; Sunday = 0) 					*/
+	int day_of_year;/**<  Day of year (0--365) 							*/
+} things_time_info;
+
```

Comment:
what resolution are you expecting here, be specific 16 bit or 32 bit. int will have varied resolution depending on the cpu alu width. on 16 bit cpu it will be 16 bits where as in 32 bit system it will be 32. so be specific about your data structure requirement